clone.go

```diff
@@ -0,0 +1,192 @@
+package poly
+
+import (
+	"errors"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+)
+
```

Comment:
Some biological background like we've done for codon optimization, seqhash etc, should be written here to explain what this is to non-biogolists.